src/extension/background-script/connectors/lndhub.ts

```diff
@@ -330,34 +330,41 @@ export default class LndHub implements Connector {
 
   async authorize() {
     const url = `${this.config.url}/auth?type=auth`;
-    const { data: authData } = await axios.post(
-      url,
-      {
-        login: this.config.login,
-        password: this.config.password,
-      },
-      {
-        headers: {
-          ...defaultHeaders,
-          "X-TS": Math.floor(Date.now() / 1000),
-          "X-VERIFY": this.generateHmacVerification(url),
+    try {
+      const { data: authData } = await axios.post(
+        url,
+        {
+          login: this.config.login,
+          password: this.config.password,
         },
-      }
-    );
+        {
+          headers: {
+            ...defaultHeaders,
+            "X-TS": Math.floor(Date.now() / 1000),
+            "X-VERIFY": this.generateHmacVerification(url),
+          },
+        }
+      );
 
-    if (authData.error || authData.errors) {
-      const error = authData.error || authData.errors;
-      const errMessage = error?.errors?.[0]?.message || error?.[0]?.message;
+      if (authData.error || authData.errors) {
+        const error = authData.error || authData.errors;
+        const errMessage = error?.errors?.[0]?.message || error?.[0]?.message;
+
+        throw new Error(errMessage);
+      }
 
-      console.error(errMessage);
-      throw new Error("API error: " + errMessage);
-    } else {
       this.refresh_token = authData.refresh_token;
       this.access_token = authData.access_token;
       this.refresh_token_created = +new Date();
       this.access_token_created = +new Date();
 
       return authData;
+    } catch (e) {
+      throw new Error(
+        `API error (${this.config.url})${
+          e instanceof Error ? `: ${e.message}` : ""
+        }`
```

Comment:
Why are we propagating this till here instead of adding this (host) in the above line itself?

Reply:
There are 2 types of errors: 
 - 5xx errors (errors on the api, not authorized, etc) throw an exception
 - Error fields in the response (that indicate the request failed) 
 
In both cases it would make sense to include the hostname in the error message.

Reply:
Ah, got it!